src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -760,30 +763,40 @@ gl_sframe drawing_classifier::predict_topk(gl_sframe data,
   // store the index in a column "rank"
   dc_predictions.add_column(
       dc_predictions["preds"].apply(argsort_prob, flex_type_enum::LIST),
-      "rank");
+      "topk_class_index");
 
   // get top k class name and store in a column "class"
-  size_t rank_idx = dc_predictions.column_index("rank");
+  size_t topk_class_index_column_idx = dc_predictions.column_index(
+    "topk_class_index");
   auto get_class_name = [=](const sframe_rows::row& row) {
-    const flex_list& rank_list = row[rank_idx];
+    const flex_list& topk_class_index_list = row[topk_class_index_column_idx];
     flex_list topk_class;
-    for (flexible_type order : rank_list) {
-      topk_class.push_back(class_labels[order]);
+    for (flexible_type class_index : topk_class_index_list) {
+      topk_class.push_back(class_labels[class_index]);
     }
     return topk_class;
   };
 
   dc_predictions.add_column(
       dc_predictions.apply(get_class_name, flex_type_enum::LIST), "class");
 
+  auto generate_ranks = [=](const sframe_rows::row& row) {
```

Comment:
yeah, this is helpful for visualization. Good.

Reply:
after I think twice, this is redundant and waste of storage. You can easily find the start of a rank interval by id % k == 0

Reply:
This is to match the output with what we used to have in 5.8.
https://github.com/apple/turicreate/blob/master/src/python/turicreate/toolkits/drawing_classifier/drawing_classifier.py#L798

Note that `id` is not like a primary key here, which is why I changed the implementation around here.

Reply:
id is the primary key for the ranking result. Thanks.

Reply:
If it's that case, you are still not doing the same thing.

Reply:
Could you explain how it's not doing the same thing?

Reply:
output id like [0001112222]; what you need to do is to change the primary key of the result by just simply divide k. That's the simplest change.

Reply:
How is that different from https://github.com/apple/turicreate/pull/2627/files/e35672aba1602f8b457bfbf1a744724a94daaf61#r346077530 ?

Reply:
I guess ` std::iota(ranks.begin(), ranks.end(), 0);` inside of your index generation.

Reply:
`generate_ranks` is generating the "rank" column whereas `add_row_number` followed by the `stack` in https://github.com/apple/turicreate/pull/2627/files/e35672aba1602f8b457bfbf1a744724a94daaf61#r346077530 is responsible for the "id" column.

Reply:
besides, the previous desgin is bad. How can I index on rows? The previous predict_topk is aligned with AC. And we should keep consistent among toolkits.

Reply:
Okay, now I know what you are doing. You stack all of them at once in the end. You should stack both rank and class (or probability) together. print out the result first and see the results.

Reply:
Why should we align with AC and not with DC in 5.8?

Reply:
because previous behavior is not good. You should at least provide a primary key for the user to index on sframe rows. I will stop here. Ask Nick.

Reply:
I think I understand this code now. I suggest simplifying it dramatically, into a single apply followed by a single stack.

For example, suppose the class labels are ["alpha", "bravo", "charlie"]. Then we want to apply each probability vector as follows:
1. Compute the arg-sort top k indices.
2. Given the k class indices in order, produce a `flex_dict` as follows:
```
flex_dict result;
for (int rank = 0; rank < k; ++k) {
  flex_int class_index = top_k_indices[rank];
  flexible_type value;
  if (output_type=="probability"]) {
    value = prob_vec[class_index];
  } else {
    value = rank;
  }
  result.emplace_back(classes[class_index], value);
}

Then unstack the result

Reply:
(But yes, I don't think we have time to design a new output format for `predict_topk`. What we're implementing here is what is used in the existing supervised learning toolkits, and is the established precedent for now)